packages/app/src/cli/services/dev/output.ts

```diff
@@ -57,29 +56,8 @@ function outputUIExtensionsURLs(extensions: UIExtension[], storeFqdn: string, ur
   }
 
   for (const extension of extensions) {
-    const heading = output.token.heading(`${extension.configuration.name} (${getHumanKey(extension.type)})`)
-    let message: string
-    switch (extension.type) {
-      case 'checkout_post_purchase': {
-        message = postPurchaseMessage(url, extension).value
-        break
-      }
-      case 'checkout_ui_extension': {
-        message = checkoutUIMessage(url, extension).value
-        break
-      }
-      case 'customer_accounts_ui_extension': {
-        message = customerAccountsUIMessage(storeFqdn, url, extension).value
-        break
-      }
-      case 'product_subscription': {
-        message = productSubscriptionMessage(url, extension).value
-        break
-      }
-      default:
-        continue
-    }
-    output.info(output.content`${heading}\n${message}\n`)
+    const message = extension.previewMessage(url, storeFqdn)
```

Comment:
this is a good example on how this project will simplify our codebase

Reply:
Yeah, agreed.  I love the changes in this file.